src/Vacuum/Vacuum_math.jl

```diff
@@ -0,0 +1,728 @@
+    p0 = legendre[1]
+    p1 = legendre[2]
+    pp = legendre[end]
+    pn = legendre[end-1]
+
+    # Chance eq.(40) 2π⅁ⁿ = G
+    gg = 2 * sqrt(π) * gamma(0.5 - n) / R
+    G = gg * pn
+
+    # Chance eq.(44)
+    # coefficient
+    grad_gg = gg / R4
+
+    if usechancebugs == false
+        # ∂Gⁿ/∂X' = dG_dX
+        aval1 = (n * (xs2 + xt2 + ζ2)*(xs2 - xt2 + ζ2) - xt2*(xt2-xs2+ζ2)) * pn
+        aval2 = (2.0 * xt * xs * (xs2-xt2+ζ2)) * pp 
+        dG_dX = grad_gg * (aval1 + aval2) / xt
+        
+        # ∂Gⁿ/∂Z' = dG_dZ
+        aval3 = (2.0 * n + 1.0) * (xs2 + xt2 + ζ2) * pn
+        aval4 = 4.0 * x_multiple * pp
+        dG_dZ = grad_gg * (aval3 + aval4) * ζ
+        
+        # Chance eq.(51) 
+        # 𝒥 ∇'𝒢ⁿ∇'ℒ = aval
+        # ∂X'/∂θ = xtp, ∂Z'/∂θ = ztp
+        aval = -xt * (ztp * dG_dX - xtp * dG_dZ)
+
+        # bval
+        bval = G
+
+        # for 𝓃⩵0,  aval0 = 𝒥 ∇'𝒢⁰∇'ℒ 
+        dG_dX0 = grad_gg * (2.0 * xt * xs * (xs2-xt2+ζ2)) * p1 - xt2*(xt2-xs2+ζ2) * p0 / xt
+        dG_dZ0 = grad_gg * ((xs2 + xt2 + ζ2) * p0 + 4.0 * x_multiple * p1) * ζ
+        aval0 = -xt * (ztp * dG_dX0 - xtp * dG_dZ0)
+    else
+        bval  = -gg*pn
+        aval1 = ( n*(xs2+xt2+ζ2)*(xt2-xs2-ζ2)+xt2*(xm2+ζ2))*pn
+        aval2 = 2.0*xt*xs*(xs2-xt2-ζ2)*pp
+        aval3 = ztp*(aval1+aval2) / xt
+        aval4 = (2.0*n+1.0)*(xp2+ζ2)*pn+4.0*xt*xs*pp
+        aval5 = xtp*(zt-zs)*aval4
+        aval6 =(aval3-aval5) / ( xt*R4)
+        aval = - xt2*aval6 * gg / (2*π)
+        aval0 = ztp*(two*xs*(zm2-ζ2)*aleg1 - xt*(xm2+ζ2)*p0)
+        aval0 = aval0 + xtp*(zt-zs)*(4.0*xt*xs*p1+(xp2+zm2)*p0)
+        aval0 = -aval0*xt / (R4*rR)
+    end
+
+    return G, aval, aval0, bval
+end
+
+#############################################################
+# Inverse Fourier transform
+#############################################################
+"""
+    gll = foranv(gil, cs, dth, twopi; m00=0, l00=0, jmax1=nothing, mth=nothing)
+
+Inverse Fourier transform from theta grid to Fourier (l) space.
+- gil: θ-grid matrix (size ≥ m00+mth, l00+jmax1)
+- cs:  transformation matrix (nths, jmax1)
+- dth: grid spacing (Float64)
+- m00, l00: starting indices (Fortran offset, usually 0)
+- jmax1: number of Fourier modes
+- mth: number of theta points
+
+Returns gll: (jmax1, jmax1) matrix.
+"""
+function foranv(gil, cs, dth ; jmax1=size(cs,2)) 
+    gll = zeros(eltype(gil), jmax1, jmax1)
+    mth=size(cs,1)
+    for l1 in 1:jmax1
+        for l2 in 1:jmax1
+            acc = zero(eltype(gil))
+            for i in 1:mth
+                acc += dth * cs[i, l2] * gil[m00 + i, l00 + l1] * 2π
+            end
+            gll[l2, l1] = acc
+        end
+    end
+    return gll
+end
+
+#############################################################
+# Utilities
+#############################################################
+
+const architecture = Sys.ARCH;
+const cpu_name = Sys.CPU_NAME;
+const threads = Sys.CPU_THREADS;
+const kernel = Sys.KERNEL;
+const username = Sys.username();
```

Comment:
What are these used for?

Also if you want the threads available to Julia itself, it's `Threads.nthreads()`